web/src/settings_profile_fields.ts

```diff
@@ -444,9 +444,17 @@ function set_up_select_field_edit_form(
     const $choice_list = $profile_field_form.find(".edit_profile_field_choices_container");
     $choice_list.off();
     $choice_list.empty();
-    const choices_data = parse_field_choices_from_field_data(
-        settings_components.select_field_data_schema.parse(JSON.parse(field.field_data)),
-    );
+    let choices_data: FieldChoice[] = [];
+    // We wrap this in a try/catch so that if the field data is malformed
+    // (e.g. invalid JSON), the edit form still opens (showing empty choices)
+    // rather than crashing. This allows the admin to recover by re-entering choices.
+    try {
```

Comment:
Is there any case that you can reproduce with invalid JSON without manually modifying the DB? If not we can remove this as well.